### ISSUE TITLE
[no-empty-function] implemented interface errors should not error

### ISSUE THREAD

**Issue by vmohir** [typescript-eslint/typescript-eslint#3841]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-empty-function": "on",
  }
}
```

```TS
class Foo implements OnInit {
  ngOnInit(): void {}
}
```

**Expected Result**

No warning because implementing an interface sometimes forces you to leave some methods empty.

**Actual Result**

The rule warns that the private constructor is useless.

**Additional Info**

`OnInit` may not be a good example because you can easily remove both the method and the implementation. But one may have an interface that has multiple methods but wants to leave some of them empty.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.30.0` |
| `@typescript-eslint/parser`        | `4.30.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.16.0` |

**Comment by bradzacher** [comment]
This rule does not use type information. It simply checks the AST. There is no way it can know about what is in your interface.

You should use a disable comment
